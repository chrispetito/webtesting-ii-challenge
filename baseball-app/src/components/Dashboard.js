import React from "react";
import Display from "./Display";

export default class Dashboard extends React.Component {
  state = {
    balls: 0,
    strikes: 0
  };

  render() {
    return (
      <div>
        <form>
          <h1 className='counter-header'>Baseball Counter</h1>
          <Display balls={this.state.balls} strikes={this.state.strikes}/>
          <div className='button-container'>
            <button className="ball-btn" onClick={this.ball}>
              Ball
            </button>
            <button className="strike-btn" onClick={this.strike}>
              Strike
            </button>
            <button className="foul-btn" onClick={this.foul}>
              Foul
            </button>
            <button className="hit-btn" onClick={this.hit}>
              Hit
            </button>
          </div>
        </form>
      </div>
    );
  }

  ball = event => {
    event.preventDefault();
    if (this.state.balls >= 3) {
      this.setState({
        balls: 0,
        strikes: 0
      });
    } else {
      this.setState({
        balls: this.state.balls + 1
      });
    }
  };

  strike = event => {
    event.preventDefault();
    if (this.state.strikes >= 2) {
      this.setState({
        balls: 0,
        strikes: 0
      });
    } else {
      this.setState({
        strikes: this.state.strikes + 1
      });
    }
  };

  foul = event => {
    event.preventDefault();
    if (this.state.strikes >= 2) {
      this.setState({
        strikes: this.state.strikes
      });
    } else {
      this.setState({
        strikes: this.state.strikes + 1
      });
    }
  };
  hit = event => {
    event.preventDefault();
    this.setState({
      balls: 0,
      strikes: 0
    });
  };
}
