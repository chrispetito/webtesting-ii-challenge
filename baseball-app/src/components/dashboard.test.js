import React from "react";
import { render, fireEvent } from "@testing-library/react";
import "@testing-library/react/cleanup-after-each";
import Dashboard from "./Dashboard";

describe("<Dashboard />", () => {
  it("Dashboard renders without crashing", () => {
    render(<Dashboard />)
  });
});
describe("Ball Button", () => {
  it("adds one ball to the count", () => {
    const { getByText } = render(<Dashboard />);
    const button = getByText(/^ball$/i);
    fireEvent.click(button);
    getByText(/^Balls: 1$/i);
  });
});
describe("Strike Button", () => {
  it("adds one strike to the count", () => {
    const { getByText } = render(<Dashboard />);
    const button = getByText(/^strike$/i);
    fireEvent.click(button);
    // console.log(button)
    getByText(/^Strikes: 1$/i);
    // expec(/^strikes: 1$/i)).toBe('Strikes: 1')
  });
});
describe("Foul Button", () => {
  it("adds one strike to the count if there are less than 2", () => {
    const { getByText } = render(<Dashboard />);
    const button = getByText(/^foul$/i);
    fireEvent.click(button);
    getByText(/^Strikes: 1$/i);
  });
});
describe("Hit Button", () => {
  it("reduces strikes and balls down to zero after a hit is recorded", () => {
    const { getByText } = render(<Dashboard />);
    const button = getByText(/^hit$/i);
    fireEvent.click(button);
    getByText(/^Balls: 0$/i);
    getByText(/^Strikes: 0$/i);
  });
});
