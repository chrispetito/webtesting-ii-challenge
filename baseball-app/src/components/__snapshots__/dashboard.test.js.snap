// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dashboard /> dashboard matches snapshot 1`] = `
<div>
  <div>
    <form>
      <h1
        class="counter-header"
      >
        Baseball Counter
      </h1>
      <div
        class="count-box"
      >
        <div>
          <h1
            class="count-number"
          >
            Balls: 
            0
          </h1>
        </div>
        <div>
          <h1
            class="count-number"
          >
            Strikes: 
            0
          </h1>
        </div>
      </div>
      <div
        class="button-container"
      >
        <button
          class="ball-btn"
        >
          Ball
        </button>
        <button
          class="strike-btn"
        >
          Strike
        </button>
        <button
          class="foul-btn"
        >
          Foul
        </button>
        <button
          class="hit-btn"
        >
          Hit
        </button>
      </div>
    </form>
  </div>
</div>
`;
